src/data/markdown/docs/02 javascript api/07 k6-http/10-set-response-callback-expectedStatuses.md

```diff
@@ -0,0 +1,53 @@
+---
+title: 'setResponseCallback( callback )'
+description: 'set responseCallback to mark responses as expected'
+---
+
+> ### 🎉 New in v0.31.0
+
+Set the response callback to be called to determine if a response was expected/successful or not.
+
+The result of this is that requests will be tagged with `expected_response` `"true"` or `"false"` and `http_req_failed` will be emitted with the reverse. This does include all redirects so if for a request only 200 is the expected response and there is 301 redirect before that, the redirect will be marked as failed as only status code 200 was marked as expected.
+
+#### Exceptions
+
+Due to implementation specifics:
+- Requests with authentication `digest` are always expected to first get 401 and then to get whatever was specified.
+- Requests with authentication `ntlm` will let a 401 status code, but also let it not be 401
```

Comment:
The `ntlm` explanation is a bit confusing. Maybe:
```suggestion
- Requests with authentication `ntlm` will not mark a 401 response as failed, but this can be overridden with `expectedStatuses`.
```

Reply:
```suggestion
- Requests with authentication `ntlm` will let a 401 status code on the first request as well as anything defined by `expectedStatuses`
```

The idea is that `digest` always wants 401 as the first response, while for NTLM it might have already been authenticated and as it is tcp connection based (the whole tcp connection is authenticated) it might not actually need to reauthenticate - so either 401 or `expectedStatuses(status)` is okay 

But yeah ... the original sentence makes no sense to me, now, as well